### DISCUSSION TITLE
Entity hierarchy when only using the bevy_ecs crate?

### DISCUSSION THREAD

**Discussion by eboatwright** [bevyengine/bevy#3955] [Q&A] [answered]
I want to use the Bevy ECS with the Macroquad rendering. So, I've got most of it done, I just wanted to know if there was already an entity hierarchy system inside the bevy_ecs crate. I couldn't find anything, so I was making sure that there wasn't a way to do it before I jump into trying to do it myself.

**Comment by alice-i-cecile** [comment, marked as answer]
What you *really* want is relations: #3742. As that is not yet implemented (and won't be for a while), I would suggest writing your own tool. The discussion in that linked issue is helpful, if somewhat overwhelming, and you can look at how `bevy_transform` does it as well.

**Reply by cart** [reply]
I'd like to emphasize that:
1. Relations may or may not be implemented as defined in #3742 (or at all). At this point we think we want something vaguely "relations shaped", but it is an active area of research. When / how that will be implemented is very tbd.
2.  You can take a dependency on `bevy_transform` and use that hierarchy directly, as long as you are willing to pull in our "transform system" too. In practice you can ignore this ... it wont incur overhead unless you add Transform components to your entities. We've also discussed breaking out hierarchy from bevy_transform, but haven't made a final call on that.

**Reply by eboatwright** [reply]
Hm. Okay I think I'll just write my own then. Because, I don't really need the positional following and stuff just mostly having access to entity id's or something that I can use to access components in other entities
